code/game/objects/items/weaponry.dm

```diff
@@ -815,6 +815,9 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 
 	var/obj/structure/table/the_table = target
 
+	if(!proximity_flag)
+		return
```

Comment:
Should you return ..()? Not sure if this can perform other actions like telekinetic ones

Reply:
No, ..() shouldn't be called here I don't believe,  all that parent does here is send a signal which should only be done if successful, this is also consistent with other cases in code.